### PR TITLE
Move base path into /api

### LINKED ISSUE #130: Move controller endpoint under "api"

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Add `spring.data.rest.base-path=/api` entry into application.yml

Resolves #130

### CODE REVIEW

File: spring-cloud-skipper-server/src/main/java/org/springframework/cloud/skipper/controller/SkipperController.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
  * @author Ilayaperumal Gopinathan
  */
 @RestController
-@RequestMapping("/")
+@RequestMapping("/api")
```
Comment (ilayaperumalg) [inline spring-cloud-skipper-server/src/main/java/org/springframework/cloud/skipper/controller/SkipperController.java]: Ideally, we can use `@BasePathAwareController` but there seems to an issue with mapping the `basePath` into underlying method request mappings.
Also, using `@RequestMapping("${spring.data.rest.base-path:/api}")` has issues outside the servlet context.

**Comment by ilayaperumalg** [comment]
Rebased.

File: spring-cloud-skipper-server/src/main/resources/application.yml [hunk 2]
```diff
@@ -7,6 +7,9 @@ server:
 spring:
   main:
     banner-mode: "off"
+  data:
+    rest:
+      base-path: /api
```
Comment (markpollack) [inline spring-cloud-skipper-server/src/main/resources/application.yml]: shoudl we make this `/api/repositories` ?

Comment (ghillert) [reply spring-cloud-skipper-server/src/main/resources/application.yml]: This is just the base-path, `repositories` will indeed be available at `/api/repositories`

**Comment by ghillert** [comment]
LGTM. Merged with commit aad93881899cd6348ebc04791dc5b5ec6eeef567

**Pull request closed without merging**